popup/search-results.js

```diff
@@ -0,0 +1,547 @@
+   * @return {Object} Response object from userstyles.org
+   */
+  function search(category) {
+    return new Promise((resolve, reject) => {
+      if (totalPages !== undefined && currentPage > totalPages) {
+        resolve({'data':[]});
+      }
+
+      const TIMEOUT = 10000;
+      const headers = {
+        'Content-type': 'application/json',
+        'Accept': '*/*'
+      };
+
+      const searchUrl = new URL('https://userstyles.org/api/v1/styles/subcategory');
+      let queryParams = 'search=' + encodeURIComponent(category);
+      queryParams += '&page=' + currentPage;
+      queryParams += '&country=NA';
+      searchUrl.search = '?' + queryParams;
+      const xhr = new XMLHttpRequest();
+      xhr.timeout = TIMEOUT;
+      xhr.onload = () => {
+        if (xhr.status === 200) {
+          const responseJson = tryJSONparse(xhr.responseText);
+          currentPage = responseJson.current_page + 1;
+          totalPages = responseJson.total_pages;
+          exhausted = (currentPage > totalPages);
+          resolve(responseJson);
+        } else {
+          exhausted = true;
+          reject(xhr.status);
+        }
+      };
+      xhr.onerror = reject;
+      xhr.open('GET', searchUrl, true);
+      for (const key of Object.keys(headers)) {
+        xhr.setRequestHeader(key, headers[key]);
+      }
+      xhr.send();
+    });
+  }
+}
+
+/**
+ * Represents the search results within the Stylus popup.
+ * @returns {Object} Includes load(), next(), and prev() methods to alter the search results.
+ */
+const SearchResults = (() => {
+  const DISPLAYED_RESULTS_PER_PAGE = 3; // Number of results to display in popup.html
+  const DELAY_AFTER_FETCHING_STYLES = 0; // Millisecs to wait before fetching next batch of search results.
+  const DELAY_BEFORE_SEARCHING_STYLES = 0; // Millisecs to wait before fetching .JSON for next search result.
+  const searchAPI = SearchUserstyles();
+  const unprocessedResults = []; // Search results not yet processed.
+  const processedResults = []; // Search results that are not installed and apply ot the page (includes 'json' field with full style).
+  const BLANK_PIXEL_DATA = 'data:image/gif;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAA' +
+                           'C1HAwCAAAAC0lEQVR42mOcXQ8AAbsBHLLDr5MAAAAASUVORK5CYII=';
+  let loading = false;
+  let tabURL; // The active tab's URL.
+  let category; // Category for the active tab's URL.
+  let currentDisplayedPage = 1; // Current page number in popup.html
+
+  // Discovery current tab's URL & USO "category" for the URL.
+  getActiveTab().then(tab => {
+    tabURL = tab.url;
+    searchAPI.getCategory(tabURL)
+      .then(resolvedCategory => {
+        category = resolvedCategory;
+        $('#searchResults-terms').textContent = category;
+      });
+  });
+
+  return {load, next, prev};
+
+  function render() {
+    $('#searchResults-list').innerHTML = ''; // Clear search results
```

Comment:
use `textContent = ''` - AMO dislikes any mentions to innerHTML so we at least avoid it in our own code